src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -386,11 +379,6 @@ public NodeStateTable<T> ToImmutableAndFree()
                     return NodeStateTable<T>.Empty;
                 }
 
-#if DEBUG
-                // If the caller requested a specific capacity, then we should have added exactly that amount of items.
-                Debug.Assert(_requestedTableCapacity == null || _requestedTableCapacity == _states.Count);
-#endif
```

Comment:
this assertion is not safe.  it's possible due to 'where' clauses that the the final amount will filter down to a lower amount than requested.

--

actually, i guess i could retrun this and change from an == to a <=.  will add that back in.